### PR TITLE
feat: Disable batch norm for densenet

### PR DESCRIPTION
**Pull request by noaonoszko**
- **Add batchnorm to simple-cnn**
- **Implement `DenseNet` with optional batch norm**
- **Steal torchvision's `_Transition`**
- **Put `_Transition`'s batch norm behind flag**
- **Steal torchvision's `_DenseBlock` and  `_DenseLayer`**
- **temp: Modify run script**
- **Add missing `_DenseLayer` methods**

### CODE REVIEW

**Comment by noaonoszko** [comment]
In favor of #14
